components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/filter/AuthParameterFilter.java

```diff
@@ -0,0 +1,190 @@
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Cache the parameters required in the authentication flow.
+ */
+public class AuthParameterFilter implements Filter {
+
+    private static final Log log = LogFactory.getLog(AuthParameterFilter.class);
+    private ServletContext context = null;
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        context = filterConfig.getServletContext();
+    }
+
+    /**
+     * @param servletRequest
+     * @param servletResponse
+     * @param filterChain
+     * @throws IOException
+     * @throws ServletException
+     */
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+
+        // Check if request contains 'sessionDataKey'.
+        if (servletRequest.getParameter("sessionDataKey") != null) {
+            if (servletRequest instanceof HttpServletRequest) {
+                // Cast ServletRequest to HttpServletRequest and get the wrapped request.
+                HttpServletRequest httpServletRequest = cacheParamsFromSessionDataKey((HttpServletRequest) servletRequest);
+                filterChain.doFilter(httpServletRequest, servletResponse);
+                return;
+            }
+        } else if (servletRequest.getParameter("sessionDataKeyConsent") != null) {
+            if (servletRequest instanceof HttpServletRequest) {
+                HttpServletRequest httpServletRequest = cacheParamsFromConsentKey((HttpServletRequest) servletRequest);
+                filterChain.doFilter(httpServletRequest, servletResponse);
+                return;
+            }
+        } else if (servletRequest.getParameter("errorKey") != null) {
+            if (servletRequest instanceof HttpServletRequest) {
+                HttpServletRequest httpServletRequest = cacheParamsFromErrorKey((HttpServletRequest) servletRequest);
+                filterChain.doFilter(httpServletRequest, servletResponse);
+                return;
+            }
+        }
+        filterChain.doFilter(servletRequest, servletResponse);
+    }
+
+    /**
+     * Cache the parameters retrieved from the API using SessionDataKey.
+     *
+     * @param servletRequest - Received servlet request.
+     * @return - HttpServletRequestWrapper with cached parameter map.
+     */
+    private HttpServletRequest cacheParamsFromSessionDataKey(HttpServletRequest servletRequest) {
+
+        String authAPIURL = buildAPIPath("sessionDataKey", servletRequest.getParameter("sessionDataKey"));
```

Comment:
Declare constants at the top and use

Reply:
- Resolved in [1f12c12](https://github.com/wso2/carbon-identity-framework/pull/4176/commits/1f12c12d9515326dc521a8dccfc37ac5f157e9d4)
- Refactored parameter retrieving code to a separate method and removed separate methods used for each key.